tests/boards/arduino/test_uno.py

```diff
@@ -48,7 +49,7 @@ def read_gpio_pin_analogue_value(self, identifier: int) -> float:
 
     def write_gpio_pin_dac_value(self, identifier: int, scaled_value: float) -> None:
         """Write a DAC value to the GPIO pin."""
-        raise NotImplementedError
+        raise NotSupportedByHardwareError
```

Comment:
This is a `NotImplemmentedError` as there is no hardware involved in this test.

Also definitely out of scope for the PR